src/modules/sub-items/components/pagination/pagination.component.js

```diff
@@ -0,0 +1,126 @@
+     *
+     * @returns {Array}
+     */
+    computePages() {
+        const { proximity, currentPage } = this.props;
+        const pagesNumber = this.getPagesNumber();
+        const pages = [];
+        let wereDots = false;
+
+        for (let i = 1; i <= pagesNumber; i++) {
+            const isFirstPage = i === 1;
+            const isLastPage = i === pagesNumber;
+            const isInRange = i >= (currentPage + 1) - proximity && i <= (currentPage + 1) + proximity;
+
+            if (isFirstPage || isLastPage || isInRange) {
+                pages.push(i);
+                wereDots = false;
+            } else if (!wereDots) {
+                pages.push('...');
+                wereDots = true;
+            }
+        }
+
+        return pages;
+    }
+
+    renderPrevious() {
+        const { currentPage, onPageChange } = this.props;
+        const previousLabel = Translator.trans(/*@Desc("Back")*/ 'pagination.previous', {}, 'sub_items');
+        const previousPage = currentPage - 1;
+        const isFirstPage = currentPage === 0;
+
+        return (
+            <PaginationBtnComponent
+                page={previousPage}
+                label={previousLabel}
+                additionalClasses="prev"
+                disabled={isFirstPage}
+                onPageChange={onPageChange}
+            />
+        );
+    }
+
+    renderNext() {
+        const { currentPage, onPageChange } = this.props;
+        const nextLabel = Translator.trans(/*@Desc("Next")*/ 'pagination.next', {}, 'sub_items');
+        const nextPage = currentPage + 1;
+        const isLastPage = currentPage === this.getPagesNumber() - 1;
+
+        return (
+            <PaginationBtnComponent
+                page={nextPage}
+                label={nextLabel}
+                additionalClasses="next"
+                disabled={isLastPage}
+                onPageChange={onPageChange}
+            />
+        );
+    }
+
+    renderPages(pages) {
+        const { currentPage, onPageChange } = this.props;
+
+        return pages.map((page, index) => {
+            if (page === '...') {
+                return <PaginationBtnComponent key={index} label="..." disabled={true} />;
+            }
+
+            const isCurrentPage = page === currentPage + 1;
+            const additionalClasses = isCurrentPage ? 'active' : '';
+            const label = '' + page;
+
+            return (
+                <PaginationBtnComponent
+                    key={page}
+                    page={page - 1}
+                    label={label}
+                    additionalClasses={additionalClasses}
+                    onPageChange={onPageChange}
+                />
+            );
+        });
+    }
+
+    render() {
+        const pages = this.computePages();
+
+        return (
+            <div className="c-pagination row justify-content-center align-items-center">
```

Comment:
I bet, the `row` CSS class should not be here.

Reply:
This is how it is styled in other places, we use `row` css class.

Reply:
Shouldn't the `row` class be applied to the nested element instead: `ul`? Because it will be the row actually.

Reply:
It can be moved to ul, I will do this.

Reply:
I moved CSS classes to `ul` element and removed `div`.